File: src/samples/php/yfrog-upload.php
<?php
    require_once('commandline.php');
    require_once('yfrog.class.php');
    
    function usage()
    {
echo <<<EOT
Usage:
    yfrog-upload.php --username USERNAME --password PASSWORD --file FILE [--tags TAGS] [--public yes | --public no]
EOT;
    }

    $username = param('username');
    $password = param('password');
    $file     = param('file');
    $tags     = param('tags');
    $public   = param('public');

    if (!$username || !$password || !$file)
    {
        usage();
        die();
    }

    $uploader = &new YfrogUploader();
    $response = $uploader->upload($file, $username, $password, $tags, $public == 'yes');
    if ($response['stat'])
        echo $response['mediaurl'];
    else
        error_log('[' . $response['code'] . '] ' . $response['msg']);


?>

File: src/samples/python/yfrog_upload.py
#!/usr/bin/env python

import yfrog
import sys

if __name__ == "__main__":

    if len(sys.argv)<4:
        print "Usage upload.py <username> <password> <filename/url> [<text>] [<source>]"
        sys.exit(1)
    u = yfrog.Uploader()
    if len(sys.argv)>=5:
        msg = sys.argv[4]
    else:
        msg = None
    if len(sys.argv)>=6:
        src = sys.argv[5]
    else:
        src = 'yfrog'

    try:
        if sys.argv[3].startswith("http://"):
            print u.uploadURL(sys.argv[3],sys.argv[1],sys.argv[2],message=msg,source=src)
        else:
            print u.uploadFile(sys.argv[3],sys.argv[1],sys.argv[2],message=msg,source=src)
    except yfrog.ServerException, e:
        print str(e)
